lib/prediction/index.js

```diff
@@ -0,0 +1,270 @@
+ * @private
+ */
+var Model = require('./model.js');
+
+/**
+ * @type {module:common/service}
+ * @private
+ */
+var Service = require('../common/service.js');
+
+/**
+ * @type {module:common/streamrouter}
+ * @private
+ */
+var streamRouter = require('../common/stream-router.js');
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+/**
+ * The [Google Prediction API](https://cloud.google.com/prediction/docs/getting-started)
+ * provides pattern-matching and machine learning capabilities. Given a set of
+ * data examples to train against, you can create applications that can perform
+ * the following tasks:
+ *
+ *   - Given a user's past viewing habits, predict what other movies or products
+ *   a user might like.
+ *   - Categorize emails as spam or non-spam.
+ *   - Analyze posted comments about your product to determine whether they have
+ *   a positive or negative tone.
+ *   - Guess how much a user might spend on a given day, given his spending
+ *   history.
+ *
+ * @resource [Hello Prediction! Example]{@link https://cloud.google.com/prediction/docs/hello_world}
+ * @resource [What is the Google Prediction API?]{@link https://cloud.google.com/prediction/docs/getting-started}
+ * @resource [Developers Guide]{@link https://cloud.google.com/prediction/docs/developer-guide}
+ *
+ * @constructor
+ * @alias module:prediction
+ *
+ * @param {object} options - [Configuration object](#/docs/?method=gcloud).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var prediction = gcloud.prediction();
+ */
+function Prediction(options) {
+  if (!(this instanceof Prediction)) {
+    options = util.normalizeArguments(this, options);
+    return new Prediction(options);
+  }
+
+  var config = {
+    baseUrl: 'https://www.googleapis.com/prediction/v1.6',
+    scopes: [
+      'https://www.googleapis.com/auth/prediction',
+      'https://www.googleapis.com/auth/devstorage.read_only'
+    ]
+  };
+
+  Service.call(this, config, options);
+}
+
+nodeutil.inherits(Prediction, Service);
+
+/**
+ * Create a trained model.
+ *
+ * You may optionally provide a {module:storage/file} as a data source to train
+ * the new model. If you have a CSV file, but haven't put it in a bucket yet,
+ * you will need to reference a bucket with {module:storage/bucket}, a file with
+ * {module:storage/file}, and upload it with {module:storage/file#upload}.
+ *
+ * @resource [Trainedmodels: insert API Documentation]{@link https://cloud.google.com/prediction/docs/reference/v1.6/trainedmodels/insert}
+ *
+ * @throws {error} If a model ID is not provided.
+ *
+ * @param {string} id - Unique name for the model. E.g. "my-model"
+ * @param {object=} options - Configuration object.
+ * @param {module:storage/file} options.data - A CSV file to use as training
+ *     data.
+ * @param {string} options.type - The type of model, `classification`
```

Comment:
Seems like we're not documenting a few options, are we trying to avoid stale docs? Or are we only documenting common use case options..?